### PR TITLE
(wip)

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: https://github.com/jenkins-infra/helpdesk/issues/2873

Tested in https://github.com/jenkins-infra/docker-inbound-agents/pull/21

### CODE REVIEW

File: vars/buildDockerAndPublishImage.groovy [hunk 1]
```diff
@@ -62,7 +62,7 @@ def call(String imageName, Map userConfig=[:]) {
               # Add folder to $PATH
               $env:Path += "$env:WORKSPACE\\.bin"
               # Add "tools" folder to $PATH
-              $env:Path += "C:\\tools"
+              $env:Path += ";C:\\tools"
```
Comment (dduportal) [inline vars/buildDockerAndPublishImage.groovy]: I might be mistaken but:

https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#withenv-set-environment-variables specifies that you can control the Path using the `withEnv()` pipeline step:

```groovy
withEnv(['PATH+TOOLS=C:/tools']) {
  powershell 'echo $env:Path'
}
```

> You may also use the syntax PATH+WHATEVER=/something to prepend /something to $PATH.

Comment (dduportal) [reply vars/buildDockerAndPublishImage.groovy]: Also, does this mean that C:\Tools is not in the current path 😱 ?

File: vars/buildDockerAndPublishImage.groovy [hunk 2]
```diff
@@ -47,53 +53,59 @@ def call(String imageName, Map userConfig=[:]) {
...
           // Logging in on the Dockerhub helps to avoid request limit from DockerHub
-          sh 'echo "${DOCKER_REGISTRY_PSW}" | "${CONTAINER_BIN}" login -u "${DOCKER_REGISTRY_USR}" --password-stdin'
+          if (operatingSystem == 'Windows') {
+            // powershell '''
+            // Set-PSDebug -Trace 1
+            // echo "DOCKER_REGISTRY_USR=${env.DOCKER_REGISTRY_USR}"
+            // $username = $env:DOCKER_REGISTRY_USR.trim()
+            // $pwd = $env:DOCKER_REGISTRY_PSW.trim()
+            // echo $pwd | docker login -u $username --password-stdin
+            // '''
```
Comment (lemeurherve) [inline vars/buildDockerAndPublishImage.groovy:63]:
```suggestion
            // '''
            sh 'echo "CONTAINER_BIN"'
            sh 'echo ${CONTAINER_BIN}'
            sh 'echo "end CONTAINER_BIN"'
```

**Comment by lemeurherve** [comment]
Superceded by https://github.com/jenkins-infra/pipeline-library/pull/385

**Pull request closed without merging**